enterprise/api_client/enterprise_catalog.py

```diff
@@ -139,6 +140,7 @@ def delete_enterprise_catalog(self, catalog_uuid):
             )
             return {}
 
+    @retry(exceptions=(ConnectionError, Timeout), tries=2, delay=4, logger=LOGGER)
```

Comment:
this whole shebang is honestly optional. Just seemed useful if we have actual temporary errors when connecting to the catalog.

Reply:
I get temporary/one time exceptions locally that this would be hugely helpful for. Do we see these in prod or staging ever? Would just make this all the more useful

Reply:
Looked back at my logs, seems like `get_course_details` is the endpoint that sometimes intermittently fails for me. I wonder if something like this could be applied there?

Reply:
Yeah we have retry logic somewhere else as well (I was trying to find a pr Brandon B wrote up that added retries somewhere but couldn't find it). This library with minimal dependencies (only depends on decorator which we already use) was just a nice and clean wrapper, so I wanted to use it.

I definitely prefer having retry logic as good code practice. However, I think in the way that we use this method it doesn't really matter since we will just rerun the job at some point anyways? So it's sort of a toss up to me about whether I actually include this or not (also I'm not sure if I want to be responsible for all the other unnecessary package upgrades via `make upgrade` breaking prod... lol)